lti_consumer/lti_consumer.py

```diff
@@ -270,6 +280,48 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
         default="",
         scope=Scope.settings
     )
+
+    # LTI 1.3 fields
+    lti_version = String(
+        display_name=_("LTI Version"),
+        scope=Scope.settings,
+        values=[
+            {"display_name": "LTI 1.1/1.2", "value": "lti_1p1"},
+            {"display_name": "LTI 1.3", "value": "lti_1p3"},
+        ],
+        default="lti_1p1",
+    )
+    lti_1p3_launch_url = String(
+        display_name=_("LTI 1.3 Launch URL"),
+        default='',
+        scope=Scope.settings
+    )
+    lti_1p3_oidc_url = String(
+        display_name=_("LTI 1.3 OIDC URL"),
+        default='',
+        scope=Scope.settings
+    )
+    lti_1p3_tool_public_key = String(
+        display_name=_("LTI 1.3 Tool Public Key"),
+        default='',
+        scope=Scope.settings
+    )
+    # Client ID and block key
+    lti_1p3_client_id = String(
+        display_name=_("LTI 1.3 Block Client ID"),
+        default=str(uuid.uuid4()),
+        scope=Scope.settings
+    )
+    # This key isn't editable, and should be regenerated
+    # for every block created (and not be carried over)
```

Comment:
You mean for each user or each block instance in the course?

Maybe `user_info` would fit here, otherwise `settings` seems fine. [ref](https://edx.readthedocs.io/projects/xblock-tutorial/en/latest/concepts/fields.html#predefined-scopes)

Reply:
When using `settings` as scope, if you duplicate the block, the key is also copied over.
I'm looking for some way to change this when copying the XBlock.

Reply:
`Scope.settings` has `Block usage` scope which I understand as independent for each instance in a course. Am I missing anything here?

Reply:
No, you're right!
But, there's 1 issue that I'm hitting: sometimes the keys from Studio and LMS differ. I'll investigate this on a follow up ticket.

Reply:
Since you're saying `default=str(uuid.uuid4())`, and this is on a class variable, the default value will be fixed _when the code is imported_. If you want a unique default for every instance, this will have to be a function call. I don't know if XBlock fields support that feature. 
The same applies to your block key below.

Reply:
@davestgermain You're right! That's precisely the issue.
I'll check if the block field API is compatible with this.